catroid/src/main/java/org/catrobat/catroid/content/Look.java

```diff
@@ -674,4 +666,307 @@ public void setHue(float hue) {
+	}
+
+	public float getYVelocityInUserInterfaceDimensionUnit() {
+		return sprite.getPhysicsProperties().getVelocity().y;
+	}
+
+	public float getX() {
+		float x = sprite.getPhysicsProperties().getX() - getWidth() / 2.0f;
+		super.setX(x);
+		return x;
+	}
+
+	@Override
+	public float getY() {
+		float y = sprite.getPhysicsProperties().getY() - getHeight() / 2.0f;
+		super.setY(y);
+		return y;
+	}
+
+	@Override
+	public float getRotation() {
+		super.setRotation((sprite.getPhysicsProperties().getDirection() % 360));
+
+		float rotation = super.getRotation();
+		float realRotation = convertStageAngleToCatroidAngle(sprite.getPhysicsProperties().getDirection() % 360);
+		if (realRotation < 0) {
+			realRotation += 360;
+		}
+
+		switch (getRotationMode()) {
+			case ROTATION_STYLE_LEFT_RIGHT_ONLY:
+				super.setRotation(0f);
+				boolean orientedRight = realRotation > 180 || realRotation == 0;
+				boolean orientedLeft = realRotation <= 180 && realRotation != 0;
+				if (((isFlipped() && orientedRight) || (!isFlipped() && orientedLeft)) && lookData != null) {
+					lookData.getTextureRegion().flip(true, false);
+				}
+				break;
+			case ROTATION_STYLE_ALL_AROUND:
+				super.setRotation(rotation);
+				break;
+			case ROTATION_STYLE_NONE:
+				super.setRotation(0f);
+				break;
+		}
+
+		return super.getRotation();
+	}
+
+	@Override
+	public void setRotation(float degrees) {
+		super.setRotation(degrees);
+		if (null != sprite.getPhysicsProperties()) {
+			sprite.getPhysicsProperties().setDirection(super.getRotation() % 360);
+		}
+	}
+
+	@Override
+	public void setScale(float scaleX, float scaleY) {
+		Vector2 oldScales = new Vector2(getScaleX(), getScaleY());
+		if (scaleX < 0.0f || scaleY < 0.0f) {
+			scaleX = 0.0f;
+			scaleY = 0.0f;
+		}
+
+		int scaleXComp = Math.round(scaleX * SCALE_FACTOR_ACCURACY);
+		int scaleYComp = Math.round(scaleY * SCALE_FACTOR_ACCURACY);
+		if (scaleXComp == Math.round(oldScales.x * SCALE_FACTOR_ACCURACY) && scaleYComp == Math.round(oldScales.y * SCALE_FACTOR_ACCURACY)) {
+			return;
+		}
+
+		super.setScale(scaleX, scaleY);
+
+		if (sprite.getPhysicsProperties() != null) {
+			PhysicsWorld physicsWorld = ProjectManager.getInstance().getSceneToPlay().getPhysicsWorld();
+			physicsWorld.changeLook(sprite.getPhysicsProperties(), this);
+			updatePhysicsObjectState(true);
+		}
+	}
+
+	public void updatePhysicsObjectState(boolean record) {
+		physicsObjectStateHandler.update(record);
+	}
+
+	public boolean isHangedUp() {
+		return physicsObjectStateHandler.isHangedUp();
+	}
+
+	public void setNonColliding(boolean nonColliding) {
+		physicsObjectStateHandler.setNonColliding(nonColliding);
+	}
+
+	public void startGlide() {
+		physicsObjectStateHandler.activateGlideTo();
+	}
+
+	public void stopGlide() {
+		physicsObjectStateHandler.deactivateGlideTo();
+	}
+
+	private interface PhysicsObjectStateCondition {
+		boolean isTrue();
+	}
+
+	private class PhysicsObjectStateHandler {
```

Comment:
since this file is already pretty huge, why not make this a standalone class and make it package private?